### PR TITLE
Milestone 4 Completion

### PR DESCRIPTION
**Pull request by ershadul1**
### What does this PR do?

- [X] Adds CategoryFilter component.
- [X] Adds filter reducer.
- [X] Adds book filtering functionalities.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @ershadul1  👋


**This project looks good. Nice work here.** 👌

However, let's make a couple of fixes and we are good to go.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { CHANGE_FILTER } from '../actions/actionTypes';
+
+const filter = (state = '', action) => {
```
Comment (mikenath223) [inline src/reducers/filter.js]: - [x] The filter reducer initial state should default to 'All'. Kindly fix this.

Comment (ershadul1) [reply src/reducers/filter.js]: Changed the initial filter state to 'All'.

**Comment by ershadul1** [comment]
- [X] The filter reducer initial state should default to 'All'

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Pull request merged**